### PR TITLE
[TES-949] Add permission checking

### PR DESCRIPTION
**Pull request by quidl**
Changes
Add default value for serverUrl
Add permission checking

Relevant issues
Resolve issue from https://issues.jenkins.io/browse/SECURITY-2845

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by duyluonganh — APPROVED** [review]

File: src/main/java/com/katalon/jenkins/plugin/ExecuteKatalonTestOpsPlan.java [hunk 1]
```diff
@@ -175,10 +176,14 @@ private String getApiKey(String credentialsId) {
       return credentials == null ? null : credentials.getSecret().getPlainText();
     }
 
+    @RequirePOST
     public FormValidation doTestConnection(@QueryParameter("serverUrl") final String url,
                                            @QueryParameter("credentialsId") final String credentialsId) {
+      Jenkins.getActiveInstance().checkPermission(Item.CONFIGURE);
+      Jenkins.getActiveInstance().checkPermission(Item.EXTENDED_READ);
```
Comment (daniel-beck) [inline src/main/java/com/katalon/jenkins/plugin/ExecuteKatalonTestOpsPlan.java]: Item/Configure implies Item/ExtendedRead, so this will never throw.

Comment (quidl) [reply src/main/java/com/katalon/jenkins/plugin/ExecuteKatalonTestOpsPlan.java]: @daniel-beck Updated, Please review again

**Comment by duyluonganh** [comment]
@daniel-beck Please help review the update of Qui again if you have time

File: src/main/java/com/katalon/jenkins/plugin/ExecuteKatalonTestOpsPlan.java [hunk 2]
```diff
@@ -175,10 +176,13 @@ private String getApiKey(String credentialsId) {
       return credentials == null ? null : credentials.getSecret().getPlainText();
     }
 
+    @RequirePOST
     public FormValidation doTestConnection(@QueryParameter("serverUrl") final String url,
                                            @QueryParameter("credentialsId") final String credentialsId) {
+      Jenkins.getActiveInstance().checkPermission(Item.CONFIGURE);
```
Comment (daniel-beck) [inline src/main/java/com/katalon/jenkins/plugin/ExecuteKatalonTestOpsPlan.java]: This is the wrong permission check: You're checking a permission that is scoped to items, globally.

1. You will prevent people from using this when they have permission to configure a specific job.
2. You will allow people who have the permission globally, even when all jobs disable permission inheritance.

See the two examples at https://www.jenkins.io/doc/developer/security/form-validation/#implementing-form-validation and choose the behavior you need for your plugin.

**Review by daniel-beck — APPROVED** [review]
Looks reasonable. Untested.

File: src/main/java/com/katalon/jenkins/plugin/ExecuteKatalonTestOpsPlan.java (hunk 2 above)
Comment (quidl) [reply src/main/java/com/katalon/jenkins/plugin/ExecuteKatalonTestOpsPlan.java]: Updated. Please review again

**Pull request merged**